src/data.rs

```diff
@@ -20,7 +23,10 @@ impl PartialEq for Data {
             (&Data::Bool(ref v0), &Data::Bool(ref v1)) => v0 == v1,
             (&Data::Vec(ref v0), &Data::Vec(ref v1)) => v0 == v1,
             (&Data::Map(ref v0), &Data::Map(ref v1)) => v0 == v1,
-            (&Data::Fun(_), &Data::Fun(_)) => bug!("Cannot compare closures"),
+            (&Data::Fun(_), &Data::Fun(_)) => {
+                bug!("Cannot compare closures");
```

Comment:
Is this a bug, or simply always false, i.e. no two closures are equal, even to themselves.

Reply:
Yeah, I wasn't really sure, so I left it as is.  I can see an argument where closures aren't reflexive (and they don't implement `PartialEq`, which is a hint that the Rust developers agree), but I can also see how users might reasonably assert that some closures are be equal.

I think we run into the halting problem, and overall asserting that they're not equal is the right thing to do, but logging a message might tell the middleware developer there's a problem and that's why the template isn't firing.  Since `bug!` is now a wrapper around `error!` I think it's okay as is.